Add @radius-project/on-call ownership for dependabot managed files

kachawla opened the pull request:
Assign the on-call team as code owners for files typically updated by dependabot to provide the ability to review and approve automated dependency updates.

Covered file types:
- Node.js: package.json, package-lock.json
- Go: go.mod, go.sum
- .NET: *.csproj
- Python: requirements.txt
- GitHub Actions: workflow files (*.yml, *.yaml)

kachawla committed:
Add @radius-project/on-call ownership for dependabot updates

Assign the on-call team as code owners for files typically updated by
dependabot to provide the ability to review and approve automated dependency
updates.

Covered file types:
- Node.js: package.json, package-lock.json
- Go: go.mod, go.sum
- .NET: *.csproj
- Python: requirements.txt
- GitHub Actions: workflow files (*.yml, *.yaml)

Signed-off-by: Karishma Chawla <kachawla@microsoft.com>

brooke-hamilton commented on .github/CODEOWNERS:
```diff
@@ -1,2 +1,21 @@
+
+# GitHub Actions workflows
+.github/workflows/*.yml   @radius-project/on-call
+.github/workflows/*.yaml  @radius-project/on-call
```
Did we had an issue in other repos in which specifying only the on-call group on selected files caused the on-call group to be the only allowed approver for those files?

For comparison, the Radius [CODEOWNERS file](https://github.com/radius-project/radius/blob/main/.github/CODEOWNERS) has entries that look like this:

```text
# Ownership for dependency-related files. Allow the on-call team to approve dependency version updates (e.g., from Dependabot, Snyk PRs)
# Files in the root directory
/go.mod @radius-project/on-call @radius-project/maintainers-radius @radius-project/approvers-radius
/go.sum @radius-project/on-call @radius-project/maintainers-radius @radius-project/approvers-radius
```

kachawla replied:
Ah yes, I completely forgot about that. I'll update it to include maintainer and approver groups as well. Thanks for catching!

kachawla replied:
updated

kachawla committed:
Include maintainers and approvers in dependabot file ownership

Add @radius-project/maintainers-docs and @radius-project/approvers-docs
to dependabot managed files to preserve existing access rights alongside
the on-call team.

Signed-off-by: Karishma Chawla <kachawla@microsoft.com>
